gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/KafkaSource.java

```diff
@@ -375,6 +378,12 @@ private void createEmptyWorkUnitsForSkippedPartitions(Map<String, List<WorkUnit>
     }
   }
 
+  private void onWorkUnitUpdate (List<String> oldTaskIds, List<WorkUnit> newWorkUnits) {
```

Comment:
Where is this method called? should it really be Source's responsibility to detect it ?